config/formats.js

```diff
@@ -493,135 +493,69 @@ let Formats = [
 		},
 	},
 	{
-		name: "[Gen 7] Partners in Crime",
-		desc: `Doubles-based metagame where both active ally Pok&eacute;mon share abilities and moves.`,
+		name: "[Gen 7] Gods and Followers",
+		desc: `The Pok&eacute;mon in the first slot is the God; the Followers must share a type with the God. If the God Pok&eacute;mon faints, the Followers are inflicted with Embargo.`,
 		threads: [
-			`&bullet; <a href="https://www.smogon.com/forums/threads/3618488/">Partners in Crime</a>`,
+			`&bullet; <a href="https://www.smogon.com/forums/threads/3589187/">Gods and Followers</a>`,
 		],
 
-		mod: 'pic',
-		gameType: 'doubles',
-		ruleset: ['[Gen 7] Doubles OU', 'Sleep Clause Mod'],
-		banlist: ['Kangaskhanite', 'Mawilite', 'Medichamite', 'Huge Power', 'Imposter', 'Normalize', 'Pure Power', 'Wonder Guard', 'Mimic', 'Sketch', 'Transform'],
-		onSwitchInPriority: 2,
-		onSwitchIn: function (pokemon) {
-			if (this.p1.active.every(ally => ally && !ally.fainted)) {
-				let p1a = this.p1.active[0], p1b = this.p1.active[1];
-				if (p1a.ability !== p1b.ability) {
-					let p1aInnate = 'ability' + p1b.ability;
-					p1a.volatiles[p1aInnate] = {id: p1aInnate, target: p1a};
-					let p1bInnate = 'ability' + p1a.ability;
-					p1b.volatiles[p1bInnate] = {id: p1bInnate, target: p1b};
-				}
-			}
-			if (this.p2.active.every(ally => ally && !ally.fainted)) {
-				let p2a = this.p2.active[0], p2b = this.p2.active[1];
-				if (p2a.ability !== p2b.ability) {
-					let p2a_innate = 'ability' + p2b.ability;
-					p2a.volatiles[p2a_innate] = {id: p2a_innate, target: p2a};
-					let p2b_innate = 'ability' + p2a.ability;
-					p2b.volatiles[p2b_innate] = {id: p2b_innate, target: p2b};
-				}
-			}
-			let ally = pokemon.side.active.find(ally => ally && ally !== pokemon && !ally.fainted);
-			if (ally && ally.ability !== pokemon.ability) {
-				// @ts-ignore
-				if (!pokemon.innate) {
-					// @ts-ignore
-					pokemon.innate = 'ability' + ally.ability;
-					// @ts-ignore
-					delete pokemon.volatiles[pokemon.innate];
-					// @ts-ignore
-					pokemon.addVolatile(pokemon.innate);
-				}
-				// @ts-ignore
-				if (!ally.innate) {
-					// @ts-ignore
-					ally.innate = 'ability' + pokemon.ability;
-					// @ts-ignore
-					delete ally.volatiles[ally.innate];
-					// @ts-ignore
-					ally.addVolatile(ally.innate);
+		mod: 'gen7',
+		ruleset: ['[Gen 7] Ubers'],
+		validateSet: function (set, teamHas) {
+			if (!teamHas.typeTable) {
+				let template = this.dex.getTemplate(set.species);
+				teamHas.typeTable = template.types;
+				let item = this.dex.getItem(set.item);
+				if (template.species === item.megaEvolves) {
+					template = this.dex.getTemplate(item.megaStone);
+					if (template.types[1] !== teamHas.typeTable[1]) teamHas.typeTable = [teamHas.typeTable[0]];
 				}
 			}
+			let problemsArray = /** @type {string[]} */ ([]);
+			let followerProblems = this.constructor('gen7ou').validateSet.call(this, set, teamHas);
```

Comment:
I think you'll find that `this.constructor('gen7ou').validateSet.call(this, set, teamHas)` is incorrect, because `this.constructor('gen7ou').validateSet` is the same function as `this.validateSet`, so it's just a useless way of writing `this.validateSet(set, teamHas);`, thus validating the whole team as Ubers.

Also, I don't think this handles Necrozma-Ultra as the god. (Then again, ROM's code looks wrong too.)

Reply:
Yeah, I meant to use TeamValidator('gen7ou') and not this.constructor('gen7ou'). it works now, i tested. fixed.

Reply:
Actually it was the use of `call` which I was complaining about, but you fixed that later anyway.

Also, I was basing my Ultra Necrozma comment on the information in the initial thread post.